docs/basic-features/environment-variables.md

```diff
@@ -124,3 +124,9 @@ This one is useful when running tests with tools like `jest` or `cypress` where
 There is a small difference between `test` environment, and both `development` and `production` that you need to bear in mind: `.env.local` won't be loaded, as you expect tests to produce the same results for everyone. This way every test execution will use same env defaults across different executions by ignoring your `.env.local` (which is intended to override the default set).
 
 > **Note**: similar to Default Environment Variables, `.env.test` file should be included in your repository, but `.env.test.local` shouldn't, as `.env*.local` are intended to be ignored through `.gitignore`.
+
+## NODE_ENV
+
+The value of `NODE_ENV` is used to determined which `.env.*` file to load when running the application. However, this value is not passed to your application code when using the `next` CLI. Instead, when running `next dev` then `process.env.NODE_ENV` is set to `development`, and when running `next start` it is set to `production`.
+
+For example, when calling `NODE_ENV=test next build`, only the environment variables in `.env.test` will be loaded, but the value of `process.env.NODE_ENV` inside your application code will be `production`. This will allow the build output to mirror your production environment exactly, aside from the environment variables in `.env.test`.
```

Comment:
This behavior will throw an error in Next 10, you cannot manually set `NODE_ENV` to something that mismatches what's provided to the application.

Reply:
Explained below 👍